### PR TITLE
Feature/devx 12 monorepo

### PR DESCRIPTION
**Pull request by ts-ron**
This PR refactors the existing repo to use node workspaces, so each sample is now using code from the `shared` dir/package. In addition a change was made to share the configuration, lint rules, and launch command, from the root directory, improving setup procedure and maintainability.

- All sample and the `shared` dir are defined as workspaces in the root `package.json`
- Moved the API URL defs into `shared/config.js` and allow override (e.g. work with EU cluster, or adapt to API gateway when it comes)
- Move all the token calls functions to the common code
- All samples now use `.env` from the root dir, saving config time
- lint / format is now performed from the root project. **I also ran it so a lot of what you see is just pretty formatting applied... sorry.**
- launching a sample is now  performed from the root project by running `SAMPLE=<sample-dir> yarn start`, no need to `cd` into a specific sample
- Tested both in a local environment and on codespaces, using both `yarn` and `npm`

TODOs before we merge:
- update README and video to reflect the above changes
- validate that this also allows us to share SCSS files, in prep fro UI alignment

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "ciam-expressjs-vanilla-samples",
+  "version": "1.0.0",
+  "main": "m.js",
```
Comment (efiwa) [inline package.json:4]: It his needed? is `m.js` produced by vite of something like that?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "ciam-expressjs-vanilla-samples",
```
Comment (efiwa) [inline package.json:2]: [minor] Maybe worth removing the `expressjs` wording to allow non-express examples in the future

**Review by efiwa — APPROVED** [review]
LGTM, (very) minor comments

File: package.json (hunk 2 above)
Comment (ts-ron) [reply package.json:2]: The intention is to have a uniform stack for a sample repo, and reflect it in the name.
We will add future repos with the other stacks and named accordingly

File: package.json (hunk 1 above)
Comment (ts-ron) [reply package.json:4]: looks like a mistake 🧐

**Pull request merged**